sentry_sdk/spotlight.py

```diff
@@ -112,9 +115,18 @@ def setup_spotlight(options):
     else:
         return None
 
-    if settings is not None and env_to_bool(
-        os.environ.get("SENTRY_SPOTLIGHT_ON_ERROR", "1")
+    if (
+        settings is not None
```

Comment:
can use `with capture_internal_exception`, we spray that liberally and it has the added benefit of being automatically disabled in sdk's own testsuite